### PR TITLE
Readded configfile, extension and resource sdk's to reference documentation

### PR DESCRIPTION
**Pull request by JCash**
With the recent changes, some documentation was left out.
This fix is a first step towards multi language sdk documentation.

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

**Comment by JCash** [comment]
Example of how it currently looks (with upcoming changes to site as well)
image

File: engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs [hunk 1]
```diff
@@ -15,36 +15,103 @@
...
+ * Configuration file access functions.
+ * The configuration file is compiled version of the [file:game.project] file.
+ *
+ * @document
+ * @language C#
+ * @name ConfigFile
+ * @namespace Dlib
+ * @path engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs
+ */
+
```
Comment (JCash) [inline engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs:30]: We now have access to the source documentation, when generating source code for other languages.

File: engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs [hunk 2]
```diff
@@ -15,36 +15,103 @@
...
+        /// </summary>
         [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi, Pack=1)]
         public struct Config
         {
         }
 
-        /*#
-        * Generated from [ref:ConfigFileGetString]
-        */
+        /// <summary>
```
Comment (JCash) [inline engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs:46]: Trying to stick with how it should look in C#...

File: engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs [hunk 3]
```diff
@@ -15,36 +15,103 @@
...
+        /// Get config value as string, returns default if the key isn't found
+        /// </summary>
+        /// <param name="config" type="HConfigFile">Config file handle</param>
+        /// <param name="key" type="const char*">Key in format section.key (.key for no section)</param>
+        /// <param name="default_value" type="const char*">Default value to return if key isn't found</param>
+        /// <returns name="value" type="const char*">found value or default value</returns>
```
Comment (JCash) [inline engine/dlib/src/cs/dmsdk/dlib/configfile_gen.cs:52]: However, the "type" attributes are "made up" by me, for keeping data around if anyone wishes to do some plugins later.

File: engine/docs/src/script_doc.py [hunk 4]
```diff
@@ -309,6 +309,8 @@ def _parse_comment(text):
             # only care for @namespace in @document comments.
             element.namespace = value
             namespace_found = True
+        elif tag == 'language':
+            element.language = value
```
Comment (JCash) [inline engine/docs/src/script_doc.py:313]: Added a "language" field to the documentation struct.

File: scripts/dmsdk/gen_cpp.py [hunk 5]
```diff
@@ -37,54 +37,101 @@ def rename_symbols(renames, line):
...
+    for (_, v) in doc.get('notes', []):
+        v = _format_doc_lines(v, indent)
+        _add_line(lines, 'note', v)
 
-    lines = [lines[0]] + ['     ' + l for l in lines[1:]]
-    return '    /*%s\n' % lines[0] + '\n'.join(lines[1:]) + '\n     */'
+    for (_, v) in doc.get('examples', []):
+        # TODO: Filter out languages that aren't C++
+        if 'objective-c' in v:
+            v = v.replace('\\@', '@')
```
Comment (JCash) [inline scripts/dmsdk/gen_cpp.py:128]: There are issues with our regex's that don't support "@" within an example... still not fixed...

File: scripts/dmsdk/gen_cpp.py [hunk 6]
```diff
@@ -219,6 +287,10 @@ def gen_cpp_header(basepath, c_header_path, out_path, info, ast, state, includes
     l('')
     l('#endif // #define DMSDK_' + basename.upper())
 
+    # Workaround, until we have proper multi language support.
+    # We attach the C documentation at the end of this file.
+    add_c_docs(docs)
```
Comment (JCash) [inline scripts/dmsdk/gen_cpp.py:292]: The current "big hack":
I adde the entire documentation of the C file into the documentation of the generated C++ file, as it's currently the one being picked up by our documentation build system.

True, multi language support comes later.

**Review by britzl — APPROVED** [review]
Looks good as far as i can tell!

**Pull request merged**